### PR TITLE
STREAMS-2399 Add alternate installation settings section

### PR DESCRIPTION
**Pull request by guillaumelamirand**
https://deploy-preview-132--streams-open-docs.netlify.app/docs/install/customize-install/

Thank you for your contribution to the Axway Streams documentation.

## Describe the changes

Enter a brief description of your changes here to communicate to the maintainers what you changed and why.

## Checklist for contributors

Before submitting this PR, please make sure:

* [ ] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [ ] You have signed the [Axway CLA](https://cla.axway.com/)
* [ ] You have verified the technical accuracy of your change
* [ ] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [ ] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

### CODE REVIEW

File: content/en/docs/install/_index.md [hunk 1]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
+export REGISTRY_USERNAME="my-service-account-client-id"
+export REGISTRY_PASSWORD="my-service-account-client-secret"
+export REGISTRY_SERVER="repository.axway.com"
 
-```sh
 kubectl create secret docker-registry "${REGISTRY_SECRET_NAME}" --docker-server="${REGISTRY_SERVER}"  --docker-username="${REGISTRY_USERNAME}" --docker-password="${REGISTRY_PASSWORD}" -n "${NAMESPACE}"
 ```
 
-## MariaDB settings
+To use your Kubernetes Secret in the registry, ad the Secret's name in the `imagePullSecrets` array. For example, add `--set imagePullSecrets[0].name="${REGISTRY_SECRET_NAME}"` to the Helm chart installation command.
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:93]: To use your Kubernetes Secret in the registry, **add** the Secret's name

File: content/en/docs/install/_index.md [hunk 2]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
 
-By default, an embedded MariaDB database is installed on your K8s cluster next to Streams. For production, we recommend that you use an externalized database instead.
+## Specific configuration for development environment
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:95]: I would not put "Specific" in prefix. Just 'Configuration for developement environment'

File: content/en/docs/install/_index.md [hunk 3]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
 
-To disable MariaDB installation, set `embeddedMariadb.enabled` to `false`. Then, choose one of the options to configure your [externalized MariaDB](#externalized-mariadb-configuration) or [embedded MariaDB](#embedded-mariadb-configuration).
+To make easily the steps with Streams, the default settings target a development environment. For production environment settings, see [dedicated section](#specific-configuration-for-production-environment).
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:97]: 'To be able to run Streams quickier/faster/quickly'

File: content/en/docs/install/_index.md [hunk 4]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
-    ```
+#### Embedded MariaDB security
 
-4. Provide information to the Streams installation. Set the following parameters:
+By default, MariaDB is configured with [TLS communication](#tls-communication) and [Transparent Data Encryption](#transparent-data-encryption-tde) enabled.
 
-    * `externalizedMariadb.host`
-    * `externalizedMariadb.port`
-    * `externalizedMariadb.rootUsername`
+To find alternate security settings, see
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:122]: To customize security settings, see ...

File: content/en/docs/install/_index.md [hunk 5]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
+## Specific configuration for production environment
 
-##### Transparent Data Encryption (TDE)
+### MariaDB settings
 
-To configure the Mariadb data-at-rest encryption, you must provide a keyfile. The keyfile must contain a 32-bit integer identifier followed by the hex-encoded encryption key separated by semicolon. For example, `<encryption_key_id>`;`<hex-encoded_encryption_key>`.
+By default, an embedded MariaDB database is installed on your K8s cluster next to Streams. For production, we recommend that you use an externalized database instead.
 
-To generate the keyfile, run the following command:
+To disable MariaDB installation, set `embeddedMariadb.enabled` to `false`.
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:209]: To disable embedded MariaDB installation.

File: content/en/docs/install/_index.md [hunk 6]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
+    export DB_USER="my-db-user"
+    export DB_NAME="streams"
+    export DB_STREAMS_USER="streams"
+    
+    mysql -h "${DB_HOST}" -P "${DB_PORT}" -u "${DB_USER}" -p -e "GRANT SELECT, INSERT, UPDATE, DELETE ON ${DB_NAME}.* TO ${DB_STREAMS_USER} REQUIRE SSL;"
+    ```
 
-* `embeddedMariadb.tls.enabled` and `embeddedMariadb.encryption.enabled` to `false`
-* `embeddedMariadb.master.extraEnvVarsSecret` and `embeddedMariadb.slave.extraEnvVarsSecret` to `null`
+4. Provide the connectivity information to the Streams installation, byt setting the [Helm parameter](/docs/install/helm-parameters/#mariadb-parameters):
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:250]: byt -> by

File: content/en/docs/install/_index.md [hunk 7]
```diff
@@ -102,377 +75,302 @@ To create your service account perform the following steps:
...
 
-{{< alert title="Note" >}}
-This option is not recommended for production environments.
-{{< /alert >}}
+    * `externalizedMariadb.host`
+    * `externalizedMariadb.port`
+    * `externalizedMariadb.rootUsername`
 
-#### Embedded MariaDB tuning
+5. Set the [Helm parameter](/docs/install/helm-parameters/#streams-parameters) `streams.serviceArgs.spring.datasource.hikari.maxLifetime` to a value (in seconds) accordingly to the `wait-timeout` value of your MariaDB database. For more information, see [Database considerations](/docs/architecture#database-considerations).
```
Comment (dominiquemongelli) [inline content/en/docs/install/_index.md:256]: to a value (in seconds) **according** to the

File: content/en/docs/install/_index.md (hunk 3 above)
Comment (guillaumelamirand) [reply content/en/docs/install/_index.md:97]: 'For a quick installation of Streams' ?

File: content/en/docs/install/_index.md [hunk 8]
```diff
@@ -7,6 +7,8 @@ hide_readingtime: true
 description: Learn how to install Streams on-premise or deploy it in your private cloud, configure a helm chart, and validate the installation.
 ---
 
+This section covers recommanded steps to install Streams. In some cases, you may want to customize your installation. For more information, see [Customize installation settings](/docs/install/customize-install).
```
Comment (andreamussap) [inline content/en/docs/install/_index.md:10]: Customer is just starting reading about the installation. First, you have to install, then customize, so no need to add this info here. The best place for it is at the end. I'd suggest you add `no_list: true` to the frontmatter of this page, so it won't list all subpages at the end, and add a last section, "Where to go next"  linking to where it makes more sense, maybe 'Customize...' and 'Helm parameters' ??

File: content/en/docs/install/helm-parameters.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Helm parameters
```
Comment (andreamussap) [inline content/en/docs/install/helm-parameters.md:2]: rename this to 'Helm parameters reference'

Comment (andreamussap) [reply content/en/docs/install/helm-parameters.md:2]: Also please add an introduction to the page, e.g., 

The Helm parameters are stored in two configuration files, `/path/values.yaml` and `/path/values-ha.yaml`. This page describe the Helm parameters that you can change to customize your configuration. (!?)

File: content/en/docs/install/customize-install.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+kubectl create secret docker-registry "${REGISTRY_SECRET_NAME}" --docker-server="${REGISTRY_SERVER}"  --docker-username="${REGISTRY_USERNAME}" --docker-password="${REGISTRY_PASSWORD}" -n "${NAMESPACE}"
+```
+
+To use your Kubernetes Secret in the registry, set the Secret's name in the `imagePullSecrets` array. For example, add `--set imagePullSecrets[0].name="${REGISTRY_SECRET_NAME}"` to the Helm chart installation command.
+
+You must also set `images.repository` accordingly to your custom registry. For more information, see [Streams parameters](/docs/install/helm-parameters#streams-parameters).
+
+## Custom embedded Mariadb security settings
+
+By default, both TLS and TDE are enabled for the embedded mariadb, however you may want to disable one or the other or both. In order to do so, please follow the one of the procedures bellow :
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:43]: Please update - "By default, both TLS and TDE encrypt modes are enabled for the embedded MariaDB server. This section shows how you can disable one or the other, or both."

File: content/en/docs/install/customize-install.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+* For enabling TLS only:
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:45]: you mean 'disable' ??  "To disable TLS only"

Please also review the next two: "To disable TDE only", "To disable both TLS and TDE security modes"

File: content/en/docs/install/customize-install.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+
+* For disabling all security features
+
+To disable MariaDB encryption **and** TLS, you must set the following [Helm parameters](/docs/install/helm-parameters/#mariadb-parameters):
+
+* `embeddedMariadb.tls.enabled` and `embeddedMariadb.encryption.enabled` to `false`
+* `embeddedMariadb.master.extraEnvVarsSecret` and `embeddedMariadb.slave.extraEnvVarsSecret` to `null`
+
+{{< alert title="Note" >}}
+Disabling security is not recommended for production environments.
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:75]: move this note up, to just behind the introduction

File: content/en/docs/install/customize-install.md [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+{{< /alert >}}
+
+## Custom embedded kafka security settings
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:78]: rename ->  **Disable** embedded kafka security settings

File: content/en/docs/install/customize-install.md [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+
+Currently, Streams works only with SASL/SCRAM authentication (using the SHA-512 hash functions) **and** TLS enabled or neither of the two.
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:80]: please update - "Streams requires you to enable both SASL/SCRAM authentication (using the SHA-512 hash functions) and TLS , or neither of the two."

File: content/en/docs/install/customize-install.md [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+* `embeddedKafka.extraVolumes` to `null`
+* `embeddedKafka.extraVolumeMounts` to `null`
+
+{{< alert title="Note" >}}
+Disabling security is not recommended for production environments.
+{{< /alert >}}
+
+## Custom externalized Mariadb security settings
+
+Depending on your MariaDB specification, you can enable either [One-Way TLS](https://mariadb.com/kb/en/securing-connections-for-client-and-server/#enabling-one-way-tls-for-mariadb-clients) or [Two-Way TLS](https://mariadb.com/kb/en/securing-connections-for-client-and-server/#enabling-two-way-tls-for-mariadb-clients) or even disabled all security feature.
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:99]: Depending on your MariaDB specification, you can enable either One-Way TLS or Two-Way TLS, or you can disabled all security features.

File: content/en/docs/install/customize-install.md [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+To disable all security features provided by kafka, you must set the following [Helm parameters](/docs/install/helm-parameters/#kafka-parameters) `externalizedKafka.auth.clientProtocol` to `plaintext`.
+
+{{< alert title="Note" >}}
+Disabling security is not recommended for production environments.
+{{< /alert >}}
+
+## Custom ingress controller settings
+
+You will find bellow some customization available for the embedded ingress controller.
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:139]: This section describes some customization available for the embedded ingress controller.

File: content/en/docs/install/customize-install.md [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+
+To disable SSL/TLS, you must set the [Helm parameters](/docs/install/helm-parameters/) `ingress.tlsenabled` to `false`.
+
+{{< alert title="Note" >}}
+Disabling security is not recommended for production environments.
+{{< /alert >}}
+
+### Custom ingress controller CORS
+
+You can enable it by setting the [Helm parameter](/docs/install/helm-parameters/) `ingress.annotations."nginx.ingress.kubernetes.io/enable-cors"` to `"true"`. For example, add the following line to the Helm Chart installation command:
```
Comment (andreamussap) [inline content/en/docs/install/customize-install.md:162]: (You can enable it by setting the) -> To enable CORS, set the

File: content/en/docs/install/_index.md (hunk 8 above)
Comment (guillaumelamirand) [reply content/en/docs/install/_index.md:10]: Actually you have to customize it while you are installing. Every customization must be done before "Helm install command" section is executed. I have updated Additionnal settings just before  Helm install command to include customization section. Let me know what you think about it.

File: content/en/docs/install/helm-parameters.md (hunk 9 above)
Comment (guillaumelamirand) [reply content/en/docs/install/helm-parameters.md:2]: ok I will do it.

File: content/en/docs/install/customize-install.md (hunk 11 above)
Comment (guillaumelamirand) [reply content/en/docs/install/customize-install.md:45]: No I mean enable. I changed the bullet points.

File: content/en/docs/install/customize-install.md (hunk 15 above)
Comment (guillaumelamirand) [reply content/en/docs/install/customize-install.md:99]: I mean enable. I updated the section.

**Comment by guillaumelamirand** [comment]
@andreamussap I have done some changes according to your comments. Could you please do a finale review ?

**Pull request merged**